apis/v1alpha2/gateway_types.go

```diff
@@ -477,6 +453,26 @@ type RouteNamespaces struct {
 	Selector *metav1.LabelSelector `json:"selector,omitempty"`
 }
 
+// RouteGroupKind indicates the group and kind of a Route resource.
+type RouteGroupKind struct {
+	// Group is the group of the Route.
+	//
+	// Support: Core
+	//
+	// +optional
+	// +kubebuilder:default=gateway.networking.k8s.io
+	// +kubebuilder:validation:MaxLength=253
+	Group *string `json:"group,omitempty"`
+
+	// Kind is the kind of the Route.
+	//
+	// Support: Core
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	Kind string `json:"kind"`
```

Comment:
Is there a status that should be set if the Kind is invalid or something?

Reply:
That's a great question, but unfortunately I don't have a good answer. This is one of the many situations where it would be unclear which controller was responsible for communicating that status. Of course every controller only knows the resources it supports, not what other implementations may support. I think the best indication of this for now would just be if this resource was missing from the `RouteParents` status altogether since no controller would have added it.

Reply:
>       // Kind MUST correspond to kinds of Routes that are compatible with the    
>       // application protocol specified in the Listener's Protocol field. If an    
>       // implementation does not support or recognize this resource type, it    
>       // SHOULD set the "ResolvedRefs" condition to false for this listener with
>       // the "InvalidRoutesRef" reason.

`ListenerRoutes` says this case should raise a status condition (on the gateway).

Reply:
Oh woops, I misinterpreted this as a question about an invalid ref from a Route to a Gateway. The answer from @jpeach is completely correct for what was actually being asked here.